poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1185,4 +1193,230 @@ public void syncPDUID(FacilitySoftwareConfig powerIQ) {
+               oldPduMap.put(FlowgateConstant.PDU_MIN_RATE_POWER, pduInfo.get(FlowgateConstant.PDU_MIN_RATE_POWER));
+               oldPduMap.put(FlowgateConstant.PDU_MAX_RATE_POWER, pduInfo.get(FlowgateConstant.PDU_MAX_RATE_POWER));
+               oldPduMap.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, pduInfo.get(FlowgateConstant.PDU_MIN_RATE_VOLTS));
+               oldPduMap.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, pduInfo.get(FlowgateConstant.PDU_MAX_RATE_VOLTS));
+               try {
+                  String newPduInfo = mapper.writeValueAsString(oldPduMap);
+                  oldJustficationfields.put(FlowgateConstant.PDU, newPduInfo);
+                  existedPduAsset.setJustificationfields(oldJustficationfields);
+               } catch (JsonProcessingException e) {
+                  logger.error("Sync pdu metadata error",e.getCause());
+               }
+               String assetSources[] = existedPduAsset.getAssetSource().split(FlowgateConstant.SPILIT_FLAG);
+               if(assetSources.length == 1) { //So far this asset source is only have powerIQ
+                  existedPduAsset.setRow(asset.getRow());
+                  existedPduAsset.setRoom(asset.getRoom());
+                  existedPduAsset.setFloor(asset.getFloor());
+                  existedPduAsset.setCity(asset.getCity());
+                  existedPduAsset.setCountry(asset.getCountry());
+                  existedPduAsset.setExtraLocation(asset.getExtraLocation());
+               }
+               //save
+               restClient.saveAssets(existedPduAsset);
+            }else {
+               asset.setAssetName(pdu.getName());
+               asset.setSerialnumber(pdu.getSerialNumber());
+               asset.setAssetSource(assetSource);
+               asset.setCategory(AssetCategory.PDU);
+               asset.setCreated(System.currentTimeMillis());
+               if(outletString != null) {
+                  pduInfo.put(FlowgateConstant.PDU_OUTLETS_FROM_POWERIQ, outletString);
+               }
+               if(inletString != null) {
+                  pduInfo.put(FlowgateConstant.PDU_INLETS_FROM_POWERIQ, inletString);
+               }
+               pduInfo.put(FlowgateConstant.PDU_ID_FROM_POWERIQ, String.valueOf(pdu.getId()));
+               String pduInfoString = null;
+               try {
+                  pduInfoString = mapper.writeValueAsString(pduInfo);
+               } catch (JsonProcessingException e) {
+                  logger.info(String.format("Sync pdu metadata error",e.getCause()));
+               }
+               if(pduInfoString != null) {
+                  HashMap<String, String> justfacationfields = new HashMap<String, String>();
+                  justfacationfields.put(FlowgateConstant.PDU, pduInfoString);
+                  asset.setJustificationfields(justfacationfields);
+               }
+               //save
+               restClient.saveAssets(asset);
+            }
+         }
+         offset += limit;
```

Comment:
you can save those 100 assets in batch. instead of save it every time.   we process by 100 each time is good enough.